addons/sale_stock/models/stock.py

```diff
@@ -72,6 +91,14 @@ class StockPicking(models.Model):
 
     sale_id = fields.Many2one(related="group_id.sale_id", string="Sales Order", store=True, readonly=False)
 
+    def _action_done(self):
+        res = super()._action_done()
+        for picking in self:
+            if picking.picking_type_code != 'outgoing' or not picking.sale_id:
```

Comment:
not picking.sale_id is enough no? We don't know about special usecase of customers.

Reply:
I changed that, instead of looping on pickings then looping on picking's moves, I loop on moves directly.
So I changed this condition for this:
```python
            if move.picking_type_id.code != 'outgoing' or not move.picking_id.sale_id or\
                move.sale_line_id or not move.quantity_done:
                continue
```
But I can changed it for that:
```python
            if not move.picking_id.sale_id or move.sale_line_id or not move.quantity_done:
                continue
```